### DISCUSSION TITLE
Debug with more than one axf

### DISCUSSION THREAD

**Discussion by xshuqun** [eclipse-cdt/cdt#682] [Q&A]
Hello All,

I need some idea on how to Debug with more than one axf file together with Jlink.
Basically, I have two projects (project_A & project_B) in one workspace and it will generate two axf (project_A.axf & project_B.axf). 
Originally using GCC, the two axf is combine together using a shell script and generates a .gdb file, afterwards it uses GDB commands for debug.

Now the problem is that, I have created a New Configuration under GDB Hardware Debugging, and set the necessary parameters, however, I don't see any place I can insert shell script so that I can combine my axf.

Anyone has any idea?

Thank you.

**Comment by jld01** [comment]
Hi @xshuqun, I suggest you create an [external tool launch configuration](https://help.eclipse.org/2023-12/index.jsp?topic=%2Forg.eclipse.platform.doc.user%2Ftasks%2Ftasks-exttools-running.htm) to launch your shell script. You can then create a [launch group](https://help.eclipse.org/2023-12/index.jsp?topic=%2Forg.eclipse.platform.doc.user%2Ftasks%2Ftasks-exttools-running.htm) to launch your shell script and launch your debug session in sequence.

Does that work for you?
